service-impl/resource-service-impl/hdfs-resource-service/src/main/java/uk/gov/gchq/palisade/resource/service/HDFSResourceService.java

```diff
@@ -86,48 +91,47 @@
     public static final String ERROR_RESOLVING_PARENTS = "Error occurred while resolving resourceParents";
 
     public static final String HADOOP_CONF_STRING = "hdfs.init.conf";
+    public static final String CACHE_IMPL_KEY = "hdfs.cache.svc";
+    public static final String CONNECTION_DETAIL_KEY = "hdfs.conn.detail";
 
     private Configuration conf;
     private FileSystem fileSystem;
 
-    private ConnectionDetailStorage connectionDetailStorage;
-
-    //===== REMOVE ONCE Issue gh-108 IS RESOLVED ============
-    public static final String USE_SHARED_STORAGE_KEY = "hdfs.init.shared.storage";
-    private boolean useSharedConnectionDetails = false;
-
-    private static ConnectionDetailStorage sharedConnectionStorage;
-
-    private ConnectionDetailStorage retrieveConnectionDetailStorage() {
-        return (useSharedConnectionDetails) ? sharedConnectionStorage : connectionDetailStorage;
-    }
-
-    private void updateSharedConnectionDetails() {
-        if (useSharedConnectionDetails) {
-            sharedConnectionStorage = this.connectionDetailStorage;
-        }
-    }
-
-    public static final String CONNECTION_DETAIL_IN_CONFIG = "hdfs.init.connection.detail";
+    private CacheService cacheService;
 
-    //=====================
+    @JsonIgnore
+    private ConnectionDetailStorage connectionDetailStorage = new ConnectionDetailStorage();
 
     public HDFSResourceService() {
     }
 
-    public HDFSResourceService(final Configuration conf, final HashMap<String, ConnectionDetail> dataFormat, final HashMap<String, ConnectionDetail> dataType) throws IOException {
-        requireNonNull(conf);
+    public HDFSResourceService(final Configuration conf, final CacheService cache) throws IOException {
+        requireNonNull(conf, "conf");
+        requireNonNull(cache, "cache");
         this.conf = conf;
         this.fileSystem = FileSystem.get(conf);
-        this.connectionDetailStorage = new ConnectionDetailStorage(dataFormat, dataType);
-        updateSharedConnectionDetails();
+        this.cacheService = cache;
     }
 
     @JsonCreator
     public HDFSResourceService(@JsonProperty("conf") final Map<String, String> conf,
-                               @JsonProperty("dataFormat") final HashMap<String, ConnectionDetail> dataFormat,
-                               @JsonProperty("dataType") final HashMap<String, ConnectionDetail> dataType) throws IOException {
-        this(createConfig(conf), dataFormat, dataType);
+                               @JsonProperty("cacheService") final CacheService cache) throws IOException {
```

Comment:
Can we change the name of the parameter to remove the need for a json property tag?